### DISCUSSION TITLE
Editor crashes using custom player controller

### DISCUSSION THREAD

**Discussion by Sheardo** [Sixze/ALS-Refactored#512] [General]
Hello,

When I use a custom player controller (parent is ALS player controller), my editor crashes.

I've tried debugging this but I am now lost. Any suggestions would be appreciated! 

```
Unhandled Exception: EXCEPTION_ACCESS_VIOLATION reading address 0x00000000000000d0

UnrealEditor_ALS!AAlsCharacter::NotifyLocomotionModeChanged() [C:\Users\james\Desktop\Test\EasySurvivalRPG V5 4.26\Plugins\ALS-Refactored-4.13\Source\ALS\Private\AlsCharacter.cpp:502]
UnrealEditor_ALS!AAlsCharacter::OnMovementModeChanged() [C:\Users\james\Desktop\Test\EasySurvivalRPG V5 4.26\Plugins\ALS-Refactored-4.13\Source\ALS\Private\AlsCharacter.cpp:480]
UnrealEditor_Engine
UnrealEditor_ALS!UAlsCharacterMovementComponent::OnMovementModeChanged() [C:\Users\james\Desktop\Test\EasySurvivalRPG V5 4.26\Plugins\ALS-Refactored-4.13\Source\ALS\Private\AlsCharacterMovementComponent.cpp:205]
UnrealEditor_Engine
UnrealEditor_Engine
UnrealEditor_Engine
UnrealEditor_Engine
UnrealEditor_Engine
UnrealEditor_Engine
UnrealEditor_ALS!UAlsCharacterMovementComponent::PerformMovement() [C:\Users\james\Desktop\Test\EasySurvivalRPG V5 4.26\Plugins\ALS-Refactored-4.13\Source\ALS\Private\AlsCharacterMovementComponent.cpp:779]
UnrealEditor_Engine
UnrealEditor_ALS!UAlsCharacterMovementComponent::ControlledCharacterMove() [C:\Users\james\Desktop\Test\EasySurvivalRPG V5 4.26\Plugins\ALS-Refactored-4.13\Source\ALS\Private\AlsCharacterMovementComponent.cpp:282]
```

**Comment by Magic-Summer** [comment]
Hi @Sheardo 
I would suggest you reinstall this plugin. The folder name `ALS-Refactored-4.13` looks like it could be an older version. I would get rid of that folder all together (when the editor is closed) and from within the `Plugin` folder follow the instructions on how to install this plugin.

As you have it on your desktop, I assume you are just trying out a downloaded project, where there could be any number of issues with said controller.

If you are now working on this project you should be working under some form of version control (ALS uses git). See [best practices > Project Setup](https://flassari.notion.site/UE-Tips-Best-Practices-3ff4c3297b414a66886c969ff741c5ba).

Generally speaking, I suggest downloading Engine Source so you get a meaningful stack trace without multiple `UnrealEditor_Engine` lines.
